vividus-plugin-web-app/src/main/java/org/vividus/proxy/steps/ProxySteps.java

```diff
@@ -138,13 +142,84 @@ public void captureRequestAndSaveURLQuery(HttpMethod httpMethod, String urlPatte
+            bddVariableContext.putVariable(scopes, variableName, queryMap(harEntry.getRequest()));
         }
     }
 
+    /**
+     * Saves the query string, body from request with given URL-pattern and response status from response
+     * into the variable with specified name and scopes.
+     * <p>
+     * This step requires proxy to be turned on.
+     * It can be done via setting properties or switching on <b>@proxy</b> metatag inside the story file.
+     * Step gets proxies log, extract from contained requests URLs and match them with URL-pattern
+     * If there is one entry, it saves the query string from request as Map of keys and values
+     * into the variable with specified name and scopes.
+     * If there weren't any calls or more than one matching requirements, HAR file with all
+     * calls will be attached to report.
+     * </p>
+     * @param httpMethod HTTP method to filter by
+     * @param urlPattern The string value of URL-pattern to filter by
+     * @param scopes The set (comma separated list of scopes e.g.: STORY, NEXT_BATCHES) of variable's scope<br>
+     * <i>Available scopes:</i>
+     * <ul>
+     * <li><b>STEP</b> - the variable will be available only within the step,
+     * <li><b>SCENARIO</b> - the variable will be available only within the scenario,
+     * <li><b>STORY</b> - the variable will be available within the whole story,
+     * <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
+     * </ul>
+     * @param variableName A variable name
+     * @throws IOException If any error happens during operation
+     */
+    @When("I capture HTTP $httpMethod request with URL pattern `$urlPattern` and save request data to $scopes "
+            + "variable `$variableName`")
+    public void captureRequestAndSaveRequestData(HttpMethod httpMethod, String urlPattern, Set<VariableScope> scopes,
+            String variableName) throws IOException
+    {
+        List<HarEntry> harEntries = checkNumberOfRequests(httpMethod, urlPattern, ComparisonRule.EQUAL_TO, 1);
+        if (harEntries.size() == 1)
+        {
+            HarEntry harEntry = harEntries.get(0);
+            HarRequest request = harEntry.getRequest();
+            HarPostData postData = request.getPostData();
+            Map<String, Map<String, String>> requestDataMap = new HashMap<>();
+            requestDataMap.put("queryMap", queryMap(request));
+            requestDataMap.put("requestBodyParamsMap", requestBodyParamsMap(postData));
+            requestDataMap.put("requestBodyMap", requestBodyMap(postData));
+            requestDataMap.put("responseStatusMap", responseStatusMap(harEntry));
+            bddVariableContext.putVariable(scopes, variableName, requestDataMap);
+        }
+    }
+
+    private Map<String, String> responseStatusMap(HarEntry harEntry)
+    {
+        int statusCode = harEntry.getResponse().getStatus();
+        return Map.of("status", String.valueOf(statusCode));
```

Comment:
why not to put status as plain value? we can not get more than one status so there is no reason to put it as a map

Reply:
We use Map<String, Map<String, String>> to save all needed information to variable, so we can't put status as plain value into this object

Reply:
Map<String, Object> ?

Reply:
If we use Map<String, Object> then we can't use varibles like ${requestData.requestBody.text} to check if it's saved correctly and we can't compare whole map, cause it's order is not guaranteed